### PR TITLE
update RELEASING.md

### PR DESCRIPTION
**Pull request by jadekler**

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -1,9 +1,26 @@
...
-6. `git checkout master && git tag v1.2.3 && git push origin v1.2.3`
-7. Update the [releases page on GitHub](https://github.com/googleapis/gax-go/releases)
+1. Determine the current release version with `git tag -l`. It should look
+   something like `vX.Y.Z`. We'll call the current version `$CV` and the new
+   version `$NV`.
+1. On master, run `git log $CV..` to list all the changes since the last
+   release.
+   a. NOTE: Some commits may pertain to only v1 or v2. Manually introspect
+   each commit to figure which occurred in v1.
+1. Edit `CHANGES.md` to include a summary of the changes.
```
Comment (noahdietz) [inline RELEASING.md]: Should these all be`1.`?

File: RELEASING.md [hunk 2]
```diff
@@ -1,9 +1,26 @@
...
+   form `v1.Y.Z`.
+   d. Push the tag: `git push origin $NV`.
+1. Update [the releases page](https://github.com/googleapis/google-cloud-go/releases)
+   with the new release, copying the contents of the CHANGES.md.
+
+# How to release v2
+
+Same process as v1, once again noting that the commit list may include v1
+commits (which should be pruned out). Note also whilst v1 tags are `v1.Y.Z`, v2
+tags are `v2.Y.Z`.
```
Comment (tbpg) [inline RELEASING.md]: Also have to update the `go.mod` with `/v2` and all local imports to use `/v2`.

Comment (jadekler) [reply RELEASING.md]: That's a one time change (when initially creating the go.mod), no? As in, you don't need to muck with the go.mod or imports each time you releasing v2 I think?

File: RELEASING.md (hunk 1 above)
Comment (jadekler) [reply RELEASING.md]: Yeah. Github automagically changes all numbers in a list to be correctly ordered. So, if you write: 1, 7, 9 it'll show up as 1, 2, 3:

1. this is 1
7. this is 7
9. this is 9

(the counting starts with whatever the first number is: if you write 3, 7, 9 it'll show up as 3, 4, 5)

File: RELEASING.md (hunk 2 above)
Comment (tbpg) [reply RELEASING.md]: Right. It kind of depends how the repo is structured. Don't want to accidentally forget. Could just be something like:

> Note: the first time you upgrade to a new version, you must update the module path in `go.mod` to include `/vN` and all local imports to use the new `/vN` path.

Comment (jadekler) [reply RELEASING.md]: Ahhh gotcha. I've added a link to the appropriate wiki section. Does that suffice?

**Pull request merged**